feat(discord): Add discord integration doc (#8160)
getsentry/sentry-docs: src/docs/product/integrations/notification-incidents/discord/index.mdx

1. In Sentry, go to **Alerts** and click "Create Alert".

2. Select "Number of Errors" or "Users Experiencing Errors" and then click "Set Conditions".

3. Configure your metric alert conditions as desired.

4. To set up Discord notifications, click "Add Action" then select your Discord server from the second dropdown.

5. Add the channel ID you'd like to send the alert to.

   The channel field must be filled in with a Discord channel ID, _not_ a channel name. To get this channel ID, copy the link to your channel and select the string of numbers at the end. For example, if the URL is `https://discord.com/channels/server-id/channel-id`, the ID is `channel-id`.

   ![Discord Channel Link](discord-copy-channel-link.png)
   ![Channel ID in Link](discord-link-channel-id.png)

6. Name and save your alert rule.

The gif below walks through how to create a metric alert that sends Discord notifications.

![Metric alert creation](discord-metric-alert-demo.gif)

### Identity Linking

Having your Discord account linked to your Sentry account allows you to resolve, archive, and assign issues directly in Discord.

> If you're the user who installed the Discord bot to your server, your identity was linked during the install process. You can use the `/link` command to verify this. The bot will tell you which Sentry account your Discord account is linked to.
> 

Reviewer: ```suggestion
When a user installs the Discord bot to their server, their identity gets linked in the process. To verify which Sentry account a Discord account is linked to, use the `/link` command.

```

> All users in any Discord server with the Sentry bot can use the `/link` command to link their accounts. If the account isn't linked yet, the bot will reply with a link to our identity linking page. Make sure you're signed into the desired Sentry account in your browser when you open the link. Click "Link with Discord" to link your accounts.

Reviewer: ```suggestion
Any user on a Discord server with the Sentry bot can link their account. To do so, use the `/link` command. The bot will reply with a link to our identity linking page. When you open the link, be sure you're signed into the Sentry account you want to link to, then click "Link with Discord".
```

![Discord identity linking page](discord-identity-link-page.png)

> To unlink your accounts, use the `/unlink` command. Follow the link and click "Unlink from Discord"

Reviewer: ```suggestion
To unlink your accounts, use the `/unlink` command. Follow the link and click "Unlink from Discord"
```

### Troubleshooting

> 

Reviewer: ```suggestion
Here are some ideas to help with troubleshooting.
```

#### Your channel ID isn't working in the alert creation wizard

If you're trying to create an alert with a Discord action and Sentry can't access the channel you've provided, please double-check the following:

> - You are giving us a channel ID and not a channel name or something else. See [Configure](#configure) for more details.
> - The channel you're trying to add is in the same server that is selected in the alert action.
> - The bot has access to the channel you're trying to use. A quick way to verify this is to check whether you can find the bot in the desired channel's member list. If you do not see the bot in the list of users, you may need to update your Discord role and/or channel permissions to allow the bot access. See [Install](#install) for more details.

Reviewer: ```suggestion
- You're giving us a channel ID and not a channel name or something else. See [Configure](#configure) for more details.
- The channel you're trying to add is in the same server as the one selected in the alert action.
- The bot has access to the channel you're trying to use. A quick way to verify this is to check whether you can find the bot in the desired channel's member list. If you don't see the bot in the list of users, you may need to update your Discord role and/or channel permissions to allow the bot access. See [Install](#install) for more details.
```
